bittensor/_subtensor/__init__.py

```diff
@@ -136,23 +139,55 @@ def __new__(
         # make sure it's wss:// or ws://
         # by default, add ws:// if neither are present
         endpoint_url: str = config.subtensor.chain_endpoint
-        if endpoint_url[0:6] != "wss://" and endpoint_url[0:5] != "ws://":
-            endpoint_url = "ws://{}".format(endpoint_url)
-        
-        substrate = SubstrateInterface(
+        endpoint_url = subtensor.format_endpoint(endpoint_url)
+
+        # Get fallback endpoints.
+        if fallback_endpoints == None:
+            if config.subtensor.fallback_endpoints != None:
+                fallback_endpoints = config.subtensor.fallback_endpoints
+            else:
+                fallback_endpoints = []
+        fallback_endpoints = [ subtensor.format_endpoint(end) for end in fallback_endpoints ]
```

Comment:
Can combine all of this into: 

```
fallback_endpoints = []
if fallback_endpoints == None and config.subtensor.fallback_endpoints != None:
               fallback_endpoints = config.subtensor.fallback_endpoints

fallback_endpoints = [ subtensor.format_endpoint(end) for end in fallback_endpoints ]
```

Reply:
+1 to this structure, I like the order bottom-up to build the scenarios we cover programmatically. We need less "resources" to understand this code. It adds readability and maintainability